pkg/lib/filesystem/ignore/ignore.go

```diff
@@ -63,9 +64,12 @@ func New(kitIgnorePaths []string, kitfile *artifact.KitFile, extraLayers ...stri
 type ignorePaths struct {
 	ignoreFileMatcher *patternmatcher.PatternMatcher
 	layers            []string
+	mu                sync.Mutex
 }
 
 func (pm *ignorePaths) Matches(path, layerPath string) (bool, error) {
+	pm.mu.Lock()
+	defer pm.mu.Unlock()
```

Comment:
`saveContentLayer` -> `packLayerToTar` -> `writeLayerToTar`
`writeLayerToTar calls `ignore.Matches` for each file With the mutex, all file traversal becomes serialized. Parallel implementation may be SLOWER than sequential due to lock contention.